project/Component/nef/CleanAPI.swift

```diff
@@ -4,17 +4,51 @@ import Foundation
 import NefCommon
 import NefModels
 import NefClean
-
 import Bow
 import BowEffects
 
 
 public extension CleanAPI {
     
-    static func clean(nefPlayground: URL) -> EnvIO<Console, nef.Error, Void> {
+    static func clean(nefPlayground: URL) -> EnvIO<ProgressReport, nef.Error, Void> {
         NefClean.Clean()
-                .nefPlayground(.init(project: nefPlayground))
-                .contramap { console in CleanEnvironment(console: console, fileSystem: MacFileSystem(), shell: MacNefPlaygroundSystem()) }
-                .mapError { e in nef.Error.compiler(info: "clean: \(e)") }
+            .nefPlayground(.init(project: nefPlayground))
+            .contramap { progressReport in
+                CleanEnvironment(
+                    progressReport: progressReport,
+                    fileSystem: MacFileSystem(),
+                    shell: MacNefPlaygroundSystem()) }
+            .mapError { e in nef.Error.compiler(info: "clean: \(e)") }
+        .foldM(
```

Comment:
it will be the same code, in other cases, just receiving the `CustomProgressDescription `

Reply:
I'll tackle this in a future PR, once I see what's the best way to do it.